app/Main.hs

```diff
@@ -46,104 +41,118 @@ runVaultIO_ action =
       Left err -> do
         die $ show err
 
+readConfig :: IO Config
+readConfig = do
+  text <- TIO.readFile "./config.toml"
+  let config = Toml.decode configCodec text
+  case config of
+    Left err -> TIO.putStrLn ("An error occurred while decoding 'config.toml':" <> Toml.prettyTomlDecodeErrors err) >> exitFailure
```

Comment:
Please print the header msg ("An error occurred ...") and the actual error in separate lines.

If they're printed in the same line, and the actual error msg spans multiple lines, it's gonna look weird.

```suggestion
    Left err -> do
      TIO.putStrLn "An error occurred while decoding 'config.toml':"
      TIO.putStrLn $ Toml.prettyTomlDecodeErrors err
      exitFailure
```